editor/plugins/asset_library_editor_plugin.cpp

```diff
@@ -705,7 +705,19 @@ void EditorAssetLibrary::_image_update(bool use_cache, bool final, const PoolByt
 
 		int len = image_data.size();
 		PoolByteArray::Read r = image_data.read();
-		Ref<Image> image = Ref<Image>(memnew(Image(r.ptr(), len)));
+		Ref<Image> image = Ref<Image>(memnew(Image));
```

Comment:
This should probably use the http content-type rather than do its own parsing.

Reply:
I'm not sure how to access that from `_image_update` function.
`HTTPRequest::response_headers` Are private members, should they made public ?

Reply:
I'm not sure how to do that because of how caching works. It's saves only raw data if I understand correctly, so when loaded from cache, my raw byte check still would be needed.

Reply:
Didn't thought about that. I would also prefer not to parse files signatures.
Getting image format from HTTP request was quite a journey :) But learnt new stuff.
Thanks for review ! I commited new changes, let me know if anything is off.